src/frontend/src/handler/create_mv.rs

```diff
@@ -166,6 +167,13 @@ pub async fn handle_create_mv(
 
     let (mut table, graph) = {
         let context = OptimizerContext::from_handler_args(handler_args);
+        if !context.with_options().is_empty() {
+            // get other useful fields by `remove`, the logic here is to reject unknown options.
+            return Err(RwError::from(ProtocolError(format!(
+                "unexpected options in WITH clause: {:?}",
+                context.with_options().keys()
```

Comment:
IMO we can have better type safety, and make it more declarative if `with_option` is a struct, rather than a hashmap of strings. But that's out of scope of this PR.

Reply:
https://github.com/risingwavelabs/risingwave/issues/13061